io/src/test/scala/fs2/io/IoSuite.scala

```diff
@@ -163,6 +164,40 @@ class IoSuite extends Fs2Suite {
         }
       }.compile.drain.map(_ => assert(true))
     }
+
+    test("different chunk sizes function correctly") {
+
+      def test(chunkSize: Int): Pipe[IO, Byte, Byte] = source => {
+        readOutputStream(chunkSize) { os =>
+          source.through(writeOutputStream(IO.delay(os), true)).compile.drain
+        }
+      }
+
+      def source(chunkSize: Int, bufferSize: Int): Stream[Pure, Byte] =
+        Stream.range(65, 75).map(_.toByte).repeat.take(chunkSize.toLong * 2).buffer(bufferSize)
+
+      forAllF { (chunkSize0: Int, bufferSize0: Int) =>
+        val chunkSize = (chunkSize0 % 512).abs + 1
+        val bufferSize = (bufferSize0 % 511).abs + 1
+
+        val src = source(chunkSize, bufferSize)
+
+        src
+          .through(text.utf8Decode)
+          .foldMonoid
+          .flatMap { expected =>
+            src
+              .through(test(chunkSize))
+              .through(text.utf8Decode)
+              .foldMonoid
+              .evalMap { actual =>
+                IO(assertEquals(actual, expected))
+              }
+          }
```

Comment:
```suggestion
        val actual = text.utf8Decode(test(chunkSize)(src)).foldMonoid
        val expectedS = text.utf8Decode(src).foldMonoid
        expectedS.flatMap { expected =>
          actualS.evalMap { actual =>
              IO(assertEquals(actual, expected))
            }
        }
```

Also, on a quick glance, can we compile both the expected and the actual stream?